kp_pre_commit_hooks/gitops-values-validation.py

```diff
@@ -0,0 +1,174 @@
+            node = destination.setdefault(key, {})
+            deep_merge(value, node)
+        else:
+            destination[key] = value
+
+    return destination
+
+
+def merge_service_values_files(service_path, instance_file):
+    """Merge values.yaml, values-env.yaml and values-env-instance.yaml files."""
+    merged_data = {}
+
+    # Base file
+    base_file = (service_path / "values.yaml").read_text(encoding="utf8")
+    merged_data.update(yaml.load(base_file))
+
+    # Environment file
+    env = "dev" if "dev" in instance_file else "prod"
+    env_file_path = service_path / f"values-{env}.yaml"
+    if env_file_path.exists():
+        env_file = env_file_path.read_text(encoding="utf8")
+        env_data = yaml.load(env_file)
+        deep_merge(env_data, merged_data)
+
+    # Instance file
+    instance_f = (service_path / instance_file).read_text(encoding="utf8")
+    instance_data = yaml.load(instance_f)
+    deep_merge(instance_data, merged_data)
+
+    return merged_data
+
+
+def find_full_error_path(error):
+    if error.parent:
+        return find_full_error_path(error.parent) + error.path
+    else:
+        return error.path
+
+
+def main():
+    if not GITOPS_DIR.exists():
+        print(f"{GITOPS_DIR} directory is missing, exiting...")
+        sys.exit(0)
+
+    error_found = False
+
+    # Iterate over direct subdirectories inside GITOPS_DIR
+    for service_path in GITOPS_DIR.glob("*/*"):
+        chart_file = service_path / "Chart.yaml"
+        value_file = service_path / "values.yaml"
+        if not (chart_file.is_file() and value_file.is_file()):
+            print(f"Chart.yaml or values.yaml file is missing in {service_path}, skipping...")
+            continue
+
+        chart_data = yaml.load(chart_file.read_text(encoding="utf8"))
+        chart_version = next(
+            (
+                dep["version"]
+                for dep in chart_data.get("dependencies", [])
+                if dep["name"] == "platform-managed-chart"
+            ),
+            None,
+        )
+
+        if not chart_version:
+            print(
+                f"Chart.yaml {chart_file} is missing platform-managed-chart dependency, skipping..."
+            )
+            continue
+
+        if not verify_values_files_schema_version(chart_version, service_path):
+            error_found = True
+            continue
+
+        schema_data = download_json_schema_for_chart_version(chart_version)
+        if not schema_data:
+            print(f"JSON schema for version {chart_version} is not supported, skipping...")
+            continue
+
+        for instance_file_path in chain(
+            service_path.glob("values-dev-*.yaml"),
+            service_path.glob("values-prod-*.yaml"),
+        ):
+            instance_file = instance_file_path.name
+            merged_values = merge_service_values_files(service_path, instance_file)
+            try:
+                validate(instance=merged_values, schema=schema_data)
```

Comment:
With validate, we fail at the first error.
It would be nicer that the dev has all the list of errors at the first instead of discovering them one by one at each run.

jsonschema seems to be able to do that with Validator and iter_errors